### PR TITLE
feature/application-data

### PR DESCRIPTION
**Pull request by tcmorris**
### *What is the problem we're trying to solve*

Implements the API as per initial discussions on [data structure](https://app.swaggerhub.com/apis/tcmorris/housingRegisterAPI/1.0.0).

### *What changes have we introduced*

- Lots of renaming to align with data
- Add properties for an application
- Updated tests

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: HousingRegisterApi/V1/Infrastructure/DynamoDbDateTimeConverter.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+using Amazon.DynamoDBv2.DataModel;
+using Amazon.DynamoDBv2.DocumentModel;
+using System;
+
+namespace HousingRegisterApi.V1.Infrastructure
+{
+    // TODO: This should go in a common NuGet package...
```
Comment (tcmorris) [inline HousingRegisterApi/V1/Infrastructure/DynamoDbDateTimeConverter.cs:7]: This is from other Hackney APIs, following same pattern for DynamoDB.

File: HousingRegisterApi/V1/Domain/Address.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+using System;
+
+namespace HousingRegisterApi.V1.Domain
+{
+    public class Address
```
Comment (angry-dan) [inline HousingRegisterApi/V1/Domain/Address.cs:5]: Should the interface that backs these classes be tied somehow to the Typescript interfaces that we have in the frontend?

File: HousingRegisterApi/V1/Infrastructure/ApplicationDbEntity.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        public string Status { get; set; }
+
+        [DynamoDBProperty(Converter = typeof(DynamoDbDateTimeConverter))]
+        public DateTime CreatedAt { get; set; }
+
+        [DynamoDBProperty(Converter = typeof(DynamoDbObjectConverter<Person>))]
+        public Person Applicant { get; set; }
+
+        [DynamoDBProperty(Converter = typeof(DynamoDbObjectListConverter<Person>))]
+        public List<Person> OtherMembers { get; set; } = new List<Person>();
```
Comment (angry-dan) [inline HousingRegisterApi/V1/Infrastructure/ApplicationDbEntity.cs:23]: This all seems quite elegant :)

**Review by angry-dan — APPROVED** [review]

File: HousingRegisterApi/V1/Domain/Address.cs (hunk 2 above)
Comment (tcmorris) [reply HousingRegisterApi/V1/Domain/Address.cs:5]: They are defined in a swagger doc, which should provide guidance to match.

**Pull request merged**